Export/Import file error for KMZ (with photos) files

**Describe the bug**
1. When exporting `KMZ` (**with photos**) files from OpenTracks, some files generate parser errors. When attempting to import these same files back into the app, the errors persist. This issue appears to be version-specific, as it doesn't occur in `v4.17.8`
2. Additionally, sharing these files to **Organic Maps** fails with a "failed to load bookmarks" error message. While Organic Maps doesn't support photo markers (expected behavior), KMZ files with photos should still load with the track data visible but without markers. Instead, these files fail completely, suggesting a file format issue rather than just the photo marker limitation.

**To Reproduce**
1. Export track data to KMZ (with photos) format
2. Attempt to import the KMZ (with photos) files back into OpenTracks
3. Alternatively, try sharing the exported KMZ (with photos) files to Organic Maps

**Error Messages**

1. When importing back to OpenTracks:
```
<file-name>.kmz: Parser error:
org.apache.harmony.xml.ExpatParser$ParseException: At line 1, column 0: no element found
```

2. When sharing to Organic Maps:
```
"Failed to load bookmarks. The file may be corrupted or defective"
```
<details>
<summary>When share KMZ to Organic Maps from OpenTracks crash log</summary>
### App information
* ID: de.dennisguse.opentracks.nightly
* Version: 6121 v4.19.1

### Device information
* Brand: Redmi
* Device: alioth
* Model: M2012K11AC
* Id: AQ3A.240912.001
* Product: alioth

### Firmware
* SDK: 35
* Release: 15
* Incremental: OS2.0.5.0.VNKCNXM

### Cause of error
```java
Exception in thread "AsyncTask #1": java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$4.done(AsyncTask.java:415)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:381)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:250)
	at java.util.concurrent.FutureTask.run(FutureTask.java:269)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1251)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:668)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'float java.lang.Float.floatValue()' on a null object reference
	at de.dennisguse.opentracks.io.file.exporter.KMLTrackExporter.getHeading(KMLTrackExporter.java:478)
	at de.dennisguse.opentracks.io.file.exporter.KMLTrackExporter.writeMarker(KMLTrackExporter.java:277)
	at de.dennisguse.opentracks.io.file.exporter.KMLTrackExporter.writeMarkers(KMLTrackExporter.java:151)
	at de.dennisguse.opentracks.io.file.exporter.KMLTrackExporter.writeTrack(KMLTrackExporter.java:116)
	at de.dennisguse.opentracks.io.file.exporter.KMZTrackExporter.writeTrack(KMZTrackExporter.java:74)
	at de.dennisguse.opentracks.data.ShareContentProvider.lambda$openFile$3(ShareContentProvider.java:216)
	at de.dennisguse.opentracks.data.ShareContentProvider$$ExternalSyntheticLambda5.writeDataToPipe(D8$$SyntheticClass:0)
	at android.content.ContentProvider$1.doInBackground(ContentProvider.java:2572)
	at android.os.AsyncTask$3.call(AsyncTask.java:394)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	... 3 more
```
</details>

**Screenshots**
![IMG_20250502_150418.jpg](https://github.com/user-attachments/assets/dfd11b99-b8a0-4e72-a46d-95a8ee1babda)

![IMG_20250502_150121.jpg](https://github.com/user-attachments/assets/03d2b986-96de-4a4f-b30b-4607a6118e6c)

![IMG_20250502_150056.jpg](https://github.com/user-attachments/assets/1cb04438-3a71-4c01-a84f-ba0bf542624c)


**Technical information**
 - Device: Poco F3
 - OS: Android 15 | HyperOS `2.0.5.0` (custom ROM)
 - OpenTracks version: `v4.19.0` or `v4.19.1` Nightly (`6121`), all from fdroid
 - OpenTracks commit id (only for nightly builds): `6121`?

--- a/src/main/java/de/dennisguse/opentracks/io/file/exporter/KMLTrackExporter.java
+++ b/src/main/java/de/dennisguse/opentracks/io/file/exporter/KMLTrackExporter.java
@@ -33,7 +33,6 @@ import java.util.ArrayList;
 import java.util.List;
 import java.util.Locale;
 import java.util.Objects;
-import java.util.Optional;
 
 import de.dennisguse.opentracks.R;
 import de.dennisguse.opentracks.data.ContentProviderUtils;
@@ -274,8 +273,7 @@ public class KMLTrackExporter implements TrackExporter {
     private void writeMarker(Marker marker, ZoneOffset zoneOffset) {
         boolean existsPhoto = MarkerUtils.buildInternalPhotoFile(context, marker.getTrackId(), marker.getPhotoUrl()) != null;
         if (marker.hasPhoto() && exportPhotos && existsPhoto) {
-            float heading = getHeading(marker.getTrackId(), marker.getPosition());
-            writePhotoOverlay(marker, heading, zoneOffset);
+            writePhotoOverlay(marker, zoneOffset);
         } else {
             writePlacemark(marker.getName(), marker.getCategory(), marker.getDescription(), marker.getPosition(), marker.getTime(), zoneOffset);
         }
@@ -429,7 +427,7 @@ public class KMLTrackExporter implements TrackExporter {
         }
     }
 
-    private void writePhotoOverlay(Marker marker, float heading, ZoneOffset zoneOffset) {
+    private void writePhotoOverlay(Marker marker, ZoneOffset zoneOffset) {
         printWriter.println("<PhotoOverlay>");
         printWriter.println("<name>" + StringUtils.formatCData(marker.getName()) + "</name>");
         printWriter.println("<description>" + StringUtils.formatCData(marker.getDescription()) + "</description>");
@@ -437,7 +435,6 @@ public class KMLTrackExporter implements TrackExporter {
         printWriter.print("<longitude>" + marker.getLongitude() + "</longitude>");
         printWriter.print("<latitude>" + marker.getLatitude() + "</latitude>");
         printWriter.print("<altitude>20</altitude>");
-        printWriter.print("<heading>" + heading + "</heading>");
         printWriter.print("<tilt>90</tilt>");
         printWriter.println("</Camera>");
         printWriter.println("<TimeStamp><when>" + getTime(zoneOffset, marker.getTime()) + "</when></TimeStamp>");
@@ -466,26 +463,6 @@ public class KMLTrackExporter implements TrackExporter {
      */
     private String getTime(ZoneOffset zoneOffset, Instant instant) {
         return StringUtils.formatDateTimeIso8601(instant, zoneOffset);
-    }
-
-    /**
-     * TODO: check if this is a useful feature (likely not).
-     * Gets the heading to a location.
-     */
-    private float getHeading(Track.Id trackId, Position position) {
-        TrackPoint.Id trackPointId = contentProviderUtils.getTrackPointId(trackId, position);
-        if (trackPointId == null) {
-            return position.bearing();
-        }
-        TrackPoint viewLocation = contentProviderUtils.getLastValidTrackPoint(trackId);
-        if (viewLocation != null) {
-            Optional<Float> bearing = viewLocation.bearingTo(position);
-            if (bearing.isPresent()) {
-                return bearing.get();
-            }
-        }
-
-        return position.bearing();
     }
 
     private static String getCoordinates(Position position, String separator) {

